src/main/java/com/google/cloud/datafusion/plugin/successfactors/source/SuccessFactorsSource.java

```diff
@@ -0,0 +1,154 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.datafusion.plugin.successfactors.source;
+
+import com.google.cloud.datafusion.plugin.successfactors.common.exception.SuccessFactorsServiceException;
+import com.google.cloud.datafusion.plugin.successfactors.common.exception.TransportException;
+import com.google.cloud.datafusion.plugin.successfactors.common.util.ExceptionParser;
+import com.google.cloud.datafusion.plugin.successfactors.common.util.ResourceConstants;
+import com.google.cloud.datafusion.plugin.successfactors.source.config.SuccessFactorsPluginConfig;
+import com.google.cloud.datafusion.plugin.successfactors.source.service.SuccessFactorsService;
+import com.google.cloud.datafusion.plugin.successfactors.source.transport.SuccessFactorsTransporter;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import org.apache.hadoop.io.LongWritable;
+
+import java.net.HttpURLConnection;
+import javax.annotation.Nullable;
+
+/**
+ * BatchSource Class
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(SuccessFactorsSource.NAME)
+@Description("Reads the SuccessFactors data which is exposed as OData services from SAP.")
+public class SuccessFactorsSource extends BatchSource<LongWritable, StructuredRecord, StructuredRecord> {
+  public static final String NAME = "SuccessFactors";
+  private final SuccessFactorsPluginConfig config;
+
+  public SuccessFactorsSource(SuccessFactorsPluginConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validatePluginParameters(failureCollector);
+
+    if (config.isSchemaBuildRequired()) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(getOutputSchema(failureCollector, false));
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(null);
+    }
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+
+  }
+
+  /**
+   * Gets the appropriate Schema basis the provided plugin parameters and also
```

Comment:
nit: based on the provided plugin...

Reply:
Updated. 0ee84795bfe46404b5b855b83136b895898b170d